src/lib/support/Pool.h

```diff
@@ -166,6 +186,40 @@ struct HeapObjectList : HeapObjectListNode
 
 } // namespace internal
 
+/// Provides iteration over active objects in a Bitmap pool.
+///
+/// The iterator is valid only if the Pool is not changed (no objects are created
+/// or released)/
```

Comment:
So this is quite unlike the ForEachActiveObject iteration, which _does_ allow releasing the object that is currently pointed to by the iterator, yes?

Reply:
I have updated this in the mean time as we seem to have unit tests that rely on this. We have unit tests to enforce that `Release` is allowed while iterating.

This is a pain though: most iterators do not like this.

Reply:
Updated the comment - this was remnant from what I had originally written of what seemed sensible to me. Apparently our internal requirements want more.